src/plugins/dashboard/public/dashboard_app/top_nav/dashboard_editing_toolbar.tsx

```diff
@@ -101,7 +101,7 @@ export function DashboardEditingToolbar() {
 
       let explicitInput: Awaited<ReturnType<typeof embeddableFactory.getExplicitInput>>;
       try {
-        explicitInput = await embeddableFactory.getExplicitInput();
+        explicitInput = await embeddableFactory.getExplicitInput(undefined, dashboard);
```

Comment:
I had to modify the `getExplicitInput` for embeddable factories to accept an optional parent - this is because, when creating the navigation container (i.e. before the container embeddable actually exists), I need to know the ID of the parent dashboard so that I can add the `current` tag to the dashboard list **even when creating a brand new navigation embeddable**:

![Screenshot 2023-07-06 at 2 43 52 PM](https://github.com/elastic/kibana/assets/8698078/36e66041-3fba-4243-9d90-cebf6a4c3c80)